core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -278,14 +286,26 @@ public void setConf(Object newConf) {
   @Override
   public List<TableIdentifier> listTables(SessionContext context, Namespace ns) {
     checkNamespaceIsValid(ns);
+    Map<String, String> queryParams = Maps.newHashMap();
+    List<TableIdentifier> tables = Lists.newArrayList();
```

Comment:
we should probably use a builder to have a `ImmutableList` here and in the other places

Reply:
will fix.

Reply:
Actually when using `ImmutableList` I get the following exception, so not sure if this is ideal
```
RESTSessionCatalog.java:305: error: [DoNotCall] This method should not be called: Always throws UnsupportedOperationException
      tables.addAll(response.identifiers());
                   ^
    (see https://errorprone.info/bugpattern/DoNotCall)
```
I see that in other areas in the `RESTSessionCatalog` that we do initialize `List` like so, so think keeping this line should be fine?
```
    private final List<ViewRepresentation> representations = Lists.newArrayList();

    List<UpdateTableRequest> tableChanges = Lists.newArrayListWithCapacity(commits.size());

```

Reply:
I think he was recommending something like:

```java
ImmutableList.Builder<TableIdentifier> result = ImmutableList.<TableIdentifier>builder();

return result.build();
```

Reply:
> I think he was recommending something like:
> 
> ```java
> ImmutableList.Builder<TableIdentifier> result = ImmutableList.<TableIdentifier>builder();
> 
> return result.build();
> ```

Yep that's exactly what I meant

Reply:
Will fix thanks